packages-exp/auth-exp/src/core/user/additional_user_info.test.ts

```diff
@@ -0,0 +1,189 @@
+/**
+ * @license
+ * Copyright 2019 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { expect } from 'chai';
+import { fromIdTokenResponse } from './additional_user_info';
+import { IdTokenResponse, IdTokenResponseKind } from '../../model/id_token';
+import { ProviderId } from '../providers';
+import { UserProfile } from '../../model/user';
+
+describe('core/user/additional_user_info', () => {
+  describe('fromIdTokenResponse', () => {
+    const userProfileWithLogin: UserProfile = {
+      login: 'scott',
+      friends: [],
+      netWorth: 5.0
+    };
+    const rawUserInfoWithLogin = JSON.stringify(userProfileWithLogin);
+    const userProfileNoLogin: UserProfile = { sample: 'data' };
+    const rawUserInfoNoLogin = JSON.stringify(userProfileNoLogin);
+
+    describe('parses federated IDP response tokens', () => {
+      it('for FacebookAdditionalUserInfo', () => {
+        const idResponse = idTokenResponse({
+          providerId: ProviderId.FACEBOOK,
+          rawUserInfo: rawUserInfoWithLogin
+        });
+        const additionalUserInfo = fromIdTokenResponse(idResponse)!;
```

Comment:
this const is only used once, recommend you collapse these two lines into const { ... } = fromIdTokenResponse()